File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int n = nums.length;
        Set<Integer> set = new HashSet<>();
        List<Integer> result = new ArrayList<>();
        
        for (int element : nums) {
            set.add(element);
        }
        
        for (int i = 1; i <= n; i++) {
            if (! set.contains(i)) {
                result.add(i);
            }
        }
        
        return result;
    }
}

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        n = len(nums)
        s = set(nums)
        result = []
        for i in range(1, n + 1):
            if not i in s:
                result.append(i)
        return result
        